### DISCUSSION TITLE
how to control the PageBreak in DynamicComponents

### DISCUSSION THREAD

**Discussion by bsdRuf** [QuestPDF/QuestPDF#603] [Q&A] [answered]
Hello, 

I have 2 diffrent types of footers. One is a one liner (sub-total) and the other is a calculation of sums, taxes and materials and services.
```
var footer = ComposeFooter(context, rows.Select(x => x.Item));
decoration.After().Element(footer);
```

I've been testing around with [Page breaks](https://www.questpdf.com/api-reference/page-break.html) but I dont get it to work.
For refrence I put the project in the post.

If the RandomGenerated items (range) fits one page it should be the "long footer" but it doesnt fit together so it takes the "short footer" and does not generate another page.

Doing a range of 31 items, does a 2 Page result with sub-total and long footer.
29/30 - just sub-total, without an additional page to show the long footer (calc)

I havent figured or found other approaches then what I tried.

Any guidiance is much appreciated, thanks in adanvce
QuestPDF has been easy to learn, is a great solution.

[QuestPDF_testing.zip](https://github.com/QuestPDF/QuestPDF/files/11966731/QuestPDF_testing.zip)

**Comment by MarcinZiabek** [comment, marked as answer]
I would simply move the "long footer" out of the Dynamic Component, and put it after the dynamic table. This way, you are certain that it is always visible after the primary content. For example:

```csharp
page.Content().Column(column =>
{
    column.Item().Dynamic(new OptimizedOrdersTable(Model.Items));
    column.Item().Element(ComposeLongFooterFunction);
});
```

**Reply by bsdRuf** [reply]
Do content flow elements work in the dynamic? Just asking because it was not mentioned on the api-reference of the content flow elements. I tried: ShowIf, ShowEntire also a fixed EnsureSpace, none came out with a new page containing the footer.

**Reply by MarcinZiabek** [reply]
There is no reason to use this type of flow control elements (ShowIf, ShowEntire, EnsureSpace, etc.). The DynamicComponent is called for each page separately, so you need to build one page at a time without any paging structures. The reason is that the available space may change between pages. 

I hope this helps.

**Reply by bsdRuf** [reply]
Maybe I missed it reading the api. Thanks a lot.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
